[JSC] Refactor `Element` in `IntlDurationFormat`

sosukesuzuki opened the pull request:
#### 831c463d4efcdcb1f6ba0679c63facfe67efc235
<pre>
[JSC] Refactor `Element` in `IntlDurationFormat`
<a href="https://bugs.webkit.org/show_bug.cgi?id=281895">https://bugs.webkit.org/show_bug.cgi?id=281895</a>

Reviewed by Yusuke Suzuki.

The `Element` struct in `IntlDurationFormat`&apos;s `collectElements` function holds the value of each
duration unit as the `m_value` field. The `m_value` field is used to pass following two pieces of
information to `NumberFormat`:

1. The sign bit
2. The value&apos;s type (`infinity`, `NaN`, or `integer`)

The type of `m_value` is always `integer`. Therefore, the `Element` struct only needs to hold the
sign bit of `m_value` rather than `m_value` itself.

This patch removes the `m_value` field from the `Element` struct and adds an `m_valueSign` field
instead.

* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::collectElements):
(JSC::IntlDurationFormat::formatToParts const):

Canonical link: <a href="https://commits.webkit.org/285992@main">https://commits.webkit.org/285992@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/013e89e43bf40e42ffaaf8489eb38e01a41a9d4f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74324 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27135 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78704 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25562 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62886 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1538 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58427 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16747 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77391 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48586 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63933 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38831 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45568 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21428 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23895 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/67461 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66977 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21775 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80220 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/73582 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1641 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/943 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66710 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1789 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63951 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65992 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16409 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9936 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8091 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/95363 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1605 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20943 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1634 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1622 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1641 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
```diff
@@ -304,7 +304,7 @@ struct Element {
     TemporalUnit m_unit;
     String m_string;
-    double m_value;
+    bool m_valueSign;
```
Can we name this `m_isValueNegative` or `m_valueSignBit`? `m_valueSign` doesn't quite indicate what true means. Also, we should probably put this after `m_type` since that packs the struct a bit smaller.

sosukesuzuki replied:
Thank you, I've addressed your comment.

sosukesuzuki committed:
[JSC] Refactor `Element` in `IntlDurationFormat`
https://bugs.webkit.org/show_bug.cgi?id=281895

Reviewed by Yusuke Suzuki.

The `Element` struct in `IntlDurationFormat`'s `collectElements` function holds the value of each
duration unit as the `m_value` field. The `m_value` field is used to pass following two pieces of
information to `NumberFormat`:

1. The sign bit
2. The value's type (`infinity`, `NaN`, or `integer`)

The type of `m_value` is always `integer`. Therefore, the `Element` struct only needs to hold the
sign bit of `m_value` rather than `m_value` itself.

This patch removes the `m_value` field from the `Element` struct and adds an `m_valueSign` field
instead.

* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::collectElements):
(JSC::IntlDurationFormat::formatToParts const):

Canonical link: https://commits.webkit.org/285992@main
